plugin/src/main/java/io/github/guegse/foreachstream/ForeachStreamCollect.java

```diff
@@ -25851,15 +25881,15 @@ public static <T0, R> R stream_limit_sorted_limit_collect(Collection<T0> input,
 			if(limit0 >= arg0) {
 				break;
 			}
-			if(limit1 >= arg1) {
-				break;
-			}
-			limit0++;
+			limit1++;
```

Comment:
Here `limit0` needs to be incremented. I think this is caused by the `i + 1` in the `Generator`'s `emitMethod` method.